AlphaWallet/TokenScriptClient/Models/UserEntryOrigin.swift

```diff
@@ -10,24 +10,40 @@ struct UserEntryOrigin {
     let originElement: XMLElement
     let xmlContext: XmlContext
     let attributeId: AttributeId
+    let bitmask: BigUInt
+    let bitShift: Int
 
-    init(originElement: XMLElement, xmlContext: XmlContext, attributeId: AttributeId, asType: OriginAsType) {
+    init(originElement: XMLElement, xmlContext: XmlContext, attributeId: AttributeId, asType: OriginAsType, bitmask: BigUInt, bitShift: Int) {
         self.originElement = originElement
         self.xmlContext = xmlContext
         self.attributeId = attributeId
         self.asType = asType
+        self.bitmask = bitmask
+        self.bitShift = bitShift
     }
 
     func extractValue(fromUserEntry userEntry: String) -> AssetInternalValue? {
         switch asType {
         case .address:
             return AlphaWallet.Address(string: userEntry.trimmed).flatMap { .address($0) }
         case .uint:
-            return BigUInt(userEntry).flatMap { .uint($0) }
-        case .utf8:
+            guard let userEntryNumber = BigUInt(userEntry, radix: 16) else { return BigUInt(userEntry).flatMap { .uint($0) } }
+            let number: BigUInt = (bitmask & userEntryNumber) >> bitShift
+            return .uint(number)
+        case .utf8, .string:
             return String(data: Data(bytes: userEntry.hexa2Bytes), encoding: .utf8).flatMap { .string($0) }
+        case .bytes:
+            guard let userEntryNumber = BigUInt(userEntry, radix: 16) else { return .bytes(Data(bytes: userEntry.hexa2Bytes)) }
```

Comment:
Can you see if `userEntry.hexa2Bytes` handles "0xF" correctly? Maybe we need to strip off the "0x" if it's there (there's a `drop0x` for that).